crates/router/src/connector/shift4/transformers.rs

```diff
@@ -120,17 +151,27 @@ pub enum CardPayment {
     CardToken(String),
 }
 
-impl<T> TryFrom<&types::RouterData<T, types::PaymentsAuthorizeData, types::PaymentsResponseData>>
-    for Shift4PaymentsRequest
+impl<T>
+    TryFrom<
+        &Shift4RouterData<
+            &types::RouterData<T, types::PaymentsAuthorizeRouterData, types::PaymentsResponseData>,
+        >,
+    > for Shift4PaymentsRequest
 {
-    type Error = Error;
+    type Error = error_stack::Report<errors::ConnectorError>;
+    // fn try_from(
+    //     item: &types::RouterData<T, types::PaymentsAuthorizeData, types::PaymentsResponseData>,
+    // ) -> Result<Self, Self::Error> {
     fn try_from(
-        item: &types::RouterData<T, types::PaymentsAuthorizeData, types::PaymentsResponseData>,
+        value: (&Shift4RouterData<
```

Comment:
can you please rename value to item
```suggestion
        item: (&Shift4RouterData<
```